x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/history/ILMHistoryStore.java

```diff
@@ -75,18 +65,12 @@ public ILMHistoryStore(Settings nodeSettings, Client client, ClusterService clus
             new BulkProcessor.Listener() {
                 @Override
                 public void beforeBulk(long executionId, BulkRequest request) {
-                    // Prior to actually performing the bulk, we should ensure the index exists, and
-                    // if we were unable to create it or it was in a bad state, we should not
-                    // attempt to index documents.
-                    try {
-                        final CompletableFuture<Boolean> indexCreated = new CompletableFuture<>();
-                        ensureHistoryIndex(client, clusterService.state(), ActionListener.wrap(indexCreated::complete,
-                            ex -> {
-                                logger.warn("failed to create ILM history store index prior to issuing bulk request", ex);
-                                indexCreated.completeExceptionally(ex);
-                            }));
-                        indexCreated.get(2, TimeUnit.MINUTES);
-                    } catch (Exception e) {
+                    if (ilmHistoryEnabled == false) {
+                        throw new ElasticsearchException("can not index ILM history items when ILM history is disabled");
+                    }
```

Comment:
I don't think we need this check here as well. 

Is this meant to catch the case when the ilm history was disabled between the time `putAsync` was called and the bulk was actually executed? If so, I think it'd be fine to allow it to go through as opposed to failing (and the next `putAsync` call will not record any history items anymore)

If we do decide to keep it I believe the exception should report the items it had not indexed.

What do you think?

Reply:
Since the `indices.lifecycle.history_index_enabled` setting is not dynamic, I don't think we need this check (since `putAsync` will always punt)

Reply:
You are right, I've removed it